data/src/main/java/io/gnosis/data/adapters/MoshiBuilder.kt

```diff
@@ -36,6 +36,14 @@ internal val settingsInfoAdapter =
         .withSubtype(SettingsInfo.EnableModule::class.java, SettingsInfoType.ENABLE_MODULE.name)
         .withSubtype(SettingsInfo.DisableModule::class.java, SettingsInfoType.DISABLE_MODULE.name)
 
+internal val unifiedEntryAdapter =
+    PolymorphicJsonAdapterFactory.of(UnifiedEntry::class.java, "type")
+        .withSubtype(UnifiedEntry.Transaction::class.java, UnifiedEntryType.TRANSACTION.name)
+        .withSubtype(UnifiedEntry.DateLabel::class.java, UnifiedEntryType.DATE_LABEL.name)
+        .withSubtype(UnifiedEntry.Label::class.java, UnifiedEntryType.LABEL.name)
+        .withSubtype(UnifiedEntry.ConflictHeader::class.java, UnifiedEntryType.CONFLICT_HEADER.name)
```

Comment:
Should we still use `UnifiedEntryType.EnumVariant.name`? would that survive a proguard code obfuscation?

Reply:
This will be fixed when we work on #1124